projects/kit/components/input-date-range/input-date-range.component.ts

```diff
@@ -191,7 +191,15 @@ export class TuiInputDateRangeComponent
     }
 
     get computedMask(): MaskitoOptions {
-        return this.activePeriod
+        /**
+         * TODO: we can delete this workaround in v4.0
+         * after solving this issue:
+         * https://github.com/taiga-family/maskito/issues/604
+         */
+        const nativeValueIsNotSynced =
+            this.textfield?.nativeFocusableElement?.value !== this.computedValue;
+
+        return this.activePeriod || nativeValueIsNotSynced
```

Comment:
Without it, it causes the following bug:
1. User selects any item from `[items]`-dropdown
2. This property `computedMask` => disable mask
3. User select any range from calendar
4. Method `onRangeChange` is triggered
5. `this.value` is updated (but native value of textfield will be asynchronously updated later)
6. This property `computedMask` => enable mask (but this native value is )
7. Maskito is initialized. It will get not up-to-date native value (item from `[items]`-dropdown). It will validate it by mask and return empty string. It will fire `input`-event. Form control is updated with empty value
8. Async operation of updating native value (from step 5) is executing